form/data_transfer_objects.rst

```diff
@@ -0,0 +1,347 @@
+make it a doctrine entity (this requires adding a primary key ``id``) and make
+the setters fluent.
+
+.. code-block:: diff
+
+    // src/Entity/Task.php
+    namespace App\Entity;
+
+    + use Symfony\Component\Validator\Constraints as Assert;
+    + use Doctrine\ORM\Mapping as ORM;
+
+    + /**
+    + * @ORM\Entity
+    + */
+    class Task
+    {
+    +   /**
+    +    * @ORM\Id()
+    +    * @ORM\GeneratedValue()
+    +    * @ORM\Column(type="integer")
+    +    */
+    +   protected $id;
+
+        /**
+    +    * @ORM\Column(type="string", length=255)
+         * @Assert\NotBlank()
+         */
+        protected $task;
+
+        /**
+    +   * @ORM\Column(type="datetime", nullable=true)
+        * @Assert\NotBlank()
+        * @Assert\Type("\DateTime")
+        */
+        protected $dueDate;
+
+    +   public function getId()
+    +   {
+    +       return $this->id;
+    +   }
+    +
+        public function getTask()
+        {
+            return $this->task;
+        }
+
+        public function setTask($task)
+        {
+            $this->task = $task;
+    +
+    +       return $this;
+        }
+
+        public function getDueDate()
+        {
+            return $this->dueDate;
+        }
+
+        public function setDueDate(\DateTime $dueDate = null)
+        {
+            $this->dueDate = $dueDate;
+    +
+    +       return $this;
+        }
+    }
+
+.. index::
+   single: Creating a Data Transfer Object
+
+Creating a Data Transfer Object
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now, create a Data Transfer Object for the ``Task`` entity using the maker:
+
+.. code-block:: terminal
+
+    $ php bin/console make:dto TaskData
+
+    The name of Entity that the DTO will be bound to:
+    > Task
+
+    Add helper extract/fill methods? (yes/no) [yes]:
+    >
+
+    Omit generation of getters/setters? (yes/no) [yes]:
+    >
+
+.. tip::
+
+    Ignore the next steps suggested by the command for now, you will generate a
+    complete CRUD with a different maker instead of a form in the next step.
+
+If you used the defaults during the dialogue, you will end up with the
+following ``TaskData`` class:
+
+.. code-block:: php
+
+    // src/Form/Data/TaskData.php
+    namespace App\Form\Data;
+
+    use App\Entity\Task;
+    use Symfony\Component\Validator\Constraints as Assert;
+
+    /**
+    * Data transfer object for Task.
+    * Add your constraints as annotations to the properties.
+    */
+    class TaskData
+    {
+        /**
+        * @Assert\NotBlank(message="This value should not be blank.", payload=null)
+        */
+        public $task;
+
+        /**
+        * @Assert\NotBlank(message="This value should not be blank.", payload=null)
+        * @Assert\Type(message="This value should be of type {{ type }}.", type="\DateTime", payload=null)
+        */
+        public $dueDate;
+
+        /**
+        * Create DTO, optionally extracting data from a model.
+        *
+        * @param Task|null $task
```

Comment:
This looks redundant to me

Reply:
Correct, I've already changed this in the generated code - updated the example.